flamedisx/inference.py

```diff
@@ -391,12 +396,67 @@ def parse_result(self, result: Minuit):
         position = {k: result.fitarg[k] for k in self.arg_names}
         return position, result.fval
 
+class NonlinearObjective(Objective):
+    """Compute limits on target parameter given nonlinear constraint
+    defined by likelihood ratio.
+    """
+    
+    # TODO: Fix callback
+    # Callback function to capture intermediate states of optimizer
+    #def fishPath(self, a, b):
+    #    self._callbackbag.append(a)
+
+    def _inner_fun_and_grad(self, params):
+        # lol who exactly is this Objective._inner_fun_and_grad(self, params)?
+        m2ll, grad, hess = Objective._inner_fun_and_grad(self, params) 
+        return m2ll - self.m2ll_best - self.t_ppf(params), grad - self.t_ppf_grad(params), hess - self.t_ppf_hess(params)
+
+    def tilt_fun(self, x):
+        return -1.*x
```

Comment:
Here we should also add the direction so that the minimizer can do both upper and lower limits.
`return -self.direction * x`